Markdownlint + WSL

Darkhost1999 opened the pull request:
In these changes, I added a copy-and-paste rough draft of the WSL instructions to the wiki with plans for later to add some more words to better communicate to users what each line of Commands is doing. Such as installing packages and compiling the code using GCC or other decorative words to help make the code blocks a little easier to read.

Speaking of making the code blocks a little easier to read I was using the markdownlint plugin inside visual code to assist in formatting the markdown. Today I re-learned about defining the languages for code blocks and observed that by utilizing them it made the markdown format "prettier" and easier to read/maintain.

These changes today help users on Windows to get up and running with future changes planned to discover ArchWSL install instructions and as mentioned clean up the documentation to put some text between the blocks for ease of readability, communication, and understanding.

Darkhost1999 committed:
markdownlint +WSL

https://github.com/DavidAnson/markdownlint/blob/v0.35.0/doc/md032.md
https://github.com/DavidAnson/markdownlint/blob/v0.35.0/doc/md040.md
https://github.com/DavidAnson/markdownlint/blob/v0.35.0/doc/md036.md
https://github.com/DavidAnson/markdownlint/blob/v0.35.0/doc/md029.md

Darkhost1999 committed:
#method-2-wsl

OCDkirby commented on install-guide.md:
````diff
@@ -51,20 +53,93 @@ sudo pacman -S --needed libunwind glfw-x11 vulkan-devel sox git cmake alsa-lib n
 
 ```sh
-cmake -B build && cmake --build build -j$(nproc)
+wsl --install
````
I believe this needs `-t Ubuntu` or something similar to install the distro from command line

Darkhost1999 replied:
You're right we shouldn't assume everyone has wsl --install -d <DistroName> set to the same value of Ubuntu by default.
I'll change it to specify Ubuntu.

Darkhost1999 replied:
Changing to `wsl --install -d Ubuntu` because d is for distro not default.
Oddly -s is for setting a default distro.
--set-default, -s <Distro>
        Sets the distribution as the default.
--terminate, -t <Distro>
        Terminates the specified distribution.

OCDkirby commented on install-guide.md:
````diff
@@ -51,20 +53,93 @@ sudo pacman -S --needed libunwind glfw-x11 vulkan-devel sox git cmake alsa-lib n
+
+```sh
+sudo apt install -y build-essential cmake libunwind-dev libglfw3-dev libvulkan-dev libsox-dev git libasound2-dev nasm g++-14
+sudo apt install -y pkgconf libasound2-plugins vainfo mesa-va-drivers
````
Why can't this be added to the command above?

Darkhost1999 replied:
I copy and posted DH's message from discord. I was wondering about that format as well. Glad to know I wasn't the only 1 questioning it. Do you think maybe WSL vs Standard Ubuntu install instructions?

Darkhost1999 replied:
Ubuntu -
 sudo apt install build-essential cmake libunwind-dev libglfw3-dev libvulkan-dev vulkan-validationlayers-dev libsox-dev git libasound2-dev nasm g++-14

Yeah, DH was showcasing Standard vs WSL specific packages. I'll add commentary

Darkhost1999 committed:
Review Changes

Learn that Break works for new lines
Make review changes
Add comments
Make Q and A more Q

Darkhost1999 commented on install-guide.md:
```diff
@@ -23,26 +25,27 @@ We provide two installation methods:
+#### Install dependencies
 
-  - **.deb-based (Ubuntu etc.)** Note: git is only needed for Ubuntu 22.04
+- **.deb-based (Ubuntu etc.)**<br>
```
Is there an alternative way to add a clean new line?
https://github.com/DavidAnson/markdownlint/blob/v0.35.0/doc/md033.md
MD033 - Inline HTML
Rationale: Raw HTML is allowed in Markdown, but this rule is included for those who want their documents to only include "pure" Markdown, or for those who are rendering Markdown documents into something other than HTML.

OCDkirby replied:
Don't make a single list item multiline with `<br/>`, it's an anti-pattern. I prefer to nest additional conditional information in the next indent level, but you could probably put it on the same level as well.

Darkhost1999 committed:
Update install-guide.md
